examples/warm_up_example/fedavg_warm_up/server.py

```diff
@@ -0,0 +1,128 @@
+    local_steps: Optional[int] = None,
+) -> Config:
+    return {
+        **make_dict_with_epochs_or_steps(local_epochs, local_steps),
+        "batch_size": batch_size,
+        "n_server_rounds": n_server_rounds,
+        "current_server_round": current_round,
+        "reporting_enabled": reporting_enabled,
+        "project_name": project_name,
+        "group_name": group_name,
+        "entity": entity,
+    }
+
+
+def main(config: Dict[str, Any], server_address: str) -> None:
+    # This function will be used to produce a config that is sent to each client to initialize their own environment
+    fit_config_fn = partial(
+        fit_config,
+        config["batch_size"],
+        config["n_server_rounds"],
+        config["reporting_config"].get("enabled", False),
+        # Note that run name is not included, it will be set in the clients
+        config["reporting_config"].get("project_name", ""),
+        config["reporting_config"].get("group_name", ""),
+        config["reporting_config"].get("entity", ""),
+        local_epochs=config.get("local_epochs"),
+        local_steps=config.get("local_steps"),
+    )
+
+    initial_parameters = get_initial_model_information()
+
+    # Server performs simple FedAveraging as its server-side optimization strategy
+    strategy = BasicFedAvg(
+        min_fit_clients=config["n_clients"],
+        min_evaluate_clients=config["n_clients"],
+        # Server waits for min_available_clients before starting FL rounds
+        min_available_clients=config["n_clients"],
+        on_fit_config_fn=fit_config_fn,
+        # We use the same fit config function, as nothing changes for eval
+        on_evaluate_config_fn=fit_config_fn,
+        fit_metrics_aggregation_fn=fit_metrics_aggregation_fn,
+        evaluate_metrics_aggregation_fn=evaluate_metrics_aggregation_fn,
+        initial_parameters=initial_parameters,
+    )
+
+    wandb_reporter = ServerWandBReporter.from_config(config)
+    client_manager = SimpleClientManager()
+    server = FlServer(client_manager, strategy, wandb_reporter)
+
+    fl.server.start_server(
+        server=server,
+        server_address=server_address,
+        config=fl.server.ServerConfig(num_rounds=config["n_server_rounds"]),
+    )
+    # Shutdown the server gracefully
+    server.shutdown()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="FL Server Main")
+    parser.add_argument(
+        "--config_path",
+        action="store",
+        type=str,
+        help="Path to configuration file.",
+        default="examples/warm_up_example/fedavg_warm_up/config.yaml",
+    )
+    parser.add_argument(
+        "--server_address",
+        action="store",
+        type=str,
+        help="Server Address to be used to communicate with the clients",
+        default="0.0.0.0:8080",
+    )
+    parser.add_argument(
+        "--seed",
+        action="store",
+        type=int,
+        help="Seed for the random number generator",
```

Comment:
Super Minor: Perhaps a similar help string as the client argument suggestion above?